frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx

```diff
@@ -0,0 +1,142 @@
+import { createPostHogWidgetNode } from 'scenes/notebooks/Nodes/NodeWrapper'
+import { FeatureFlagBasicType, NotebookNodeType, Survey, SurveyQuestionType } from '~/types'
+import { BindLogic, useActions, useValues } from 'kea'
+import { IconFlag, IconSurveys } from 'lib/lemon-ui/icons'
+import { LemonButton, LemonDivider } from '@posthog/lemon-ui'
+import { urls } from 'scenes/urls'
+import { LemonSkeleton } from 'lib/lemon-ui/LemonSkeleton'
+import { notebookNodeLogic } from './notebookNodeLogic'
+import { NotebookNodeViewProps } from '../Notebook/utils'
+import { buildFlagContent } from './NotebookNodeFlag'
+import { defaultSurveyAppearance, surveyLogic } from 'scenes/surveys/surveyLogic'
+import { StatusTag } from 'scenes/surveys/Surveys'
+import { SurveyResult } from 'scenes/surveys/SurveyView'
+import { SurveyAppearance } from 'scenes/surveys/SurveyAppearance'
+import { SurveyReleaseSummary } from 'scenes/surveys/Survey'
+import api from 'lib/api'
+
+const Component = (props: NotebookNodeViewProps<NotebookNodeSurveyAttributes>): JSX.Element => {
+    const { id } = props.node.attrs
+    const { survey, surveyLoading, hasTargetingFlag } = useValues(surveyLogic({ id }))
+    const { expanded, nextNode } = useValues(notebookNodeLogic)
+    const { insertAfter } = useActions(notebookNodeLogic)
+
+    return (
+        <div>
+            <BindLogic logic={surveyLogic} props={{ id }}>
+                <div className="flex items-center gap-2 p-3">
+                    <IconSurveys className="text-lg" />
+                    {surveyLoading ? (
+                        <LemonSkeleton className="h-6 flex-1" />
+                    ) : (
+                        <>
+                            <span className="flex-1 font-semibold truncate">{survey.name}</span>
+                            {/* survey has to exist in notebooks */}
```

Comment:
Might be worth adding an error state in cases when the survey is missing. To prevent the need for casting or an optional object